FindBaselibs.cmake

```diff
@@ -1,89 +1,71 @@
-
-set (INC_HDF5 ${BASEDIR}/include/hdf5)
-set (INC_NETCDF ${BASEDIR}/include/netcdf)
-set (INC_HDF ${BASEDIR}/include/hdf)
-set (INC_ESMF ${BASEDIR}/include/esmf)
-
-find_package(GFTL REQUIRED)
-find_package(GFTL_SHARED QUIET)
-find_package(FARGPARSE QUIET)
-
-find_package(FLAP REQUIRED)
-
-if (APPLE)
-  if (NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-    execute_process (COMMAND ${CMAKE_C_COMPILER} --print-file-name=libgcc.a OUTPUT_VARIABLE libgcc OUTPUT_STRIP_TRAILING_WHITESPACE)
-  endif ()
-  execute_process (COMMAND ${CMAKE_CXX_COMPILER} --print-file-name=libstdc++.dylib OUTPUT_VARIABLE stdcxx OUTPUT_STRIP_TRAILING_WHITESPACE)
-else ()
-  execute_process (COMMAND ${CMAKE_CXX_COMPILER} --print-file-name=libstdc++.so OUTPUT_VARIABLE stdcxx OUTPUT_STRIP_TRAILING_WHITESPACE)
-endif ()
-
-# For OS X - use the gcc stdc++ library - not clang
-#find_library (STDCxx
-#  stdc++
-#  HINTS "/opt/local/lib/gcc5"
-#  )
-
-# We must statically link ESMF on Apple due mainly to an issue with how Baselibs is built.
-# Namely, the esmf dylib libraries end up with the full *build* path on Darwin (which is in 
-# src/esmf/lib/libO...) But we copy the dylib to $BASEDIR/lib. Thus, DYLD_LIBRARY_PATH gets
-# hosed. yay.
-
-if (APPLE)
-   set (ESMF_LIBRARY ${BASEDIR}/lib/libesmf.a)
-else ()
-   set (ESMF_LIBRARY esmf_fullylinked)
-endif ()
-
-#find_package (NetCDF REQUIRED COMPONENTS Fortran)
-#set (INC_NETCDF ${NETCDF_INCLUDE_DIRS})
-
-set (NETCDF_LIBRARIES ${NETCDF_LIBRARIES_OLD})
-set (ESMF_LIBRARIES ${ESMF_LIBRARY} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} ${MPI_CXX_LIBRARIES} ${stdcxx} ${libgcc})
+# Find NetCDF
+find_package(NetCDF REQUIRED COMPONENTS C Fortran)
+# Set non-standard expected variables
+set(INC_NETCDF ${NETCDF_INCLUDE_DIRS})
+set(LIB_NETCDF ${NETCDF_LIBRARIES})
+
+# If BASEDIR exists, set the expected HDF and HDF5 variables
+if(EXISTS ${BASEDIR}/include/hdf5)
+  set (INC_HDF5 ${BASEDIR}/include/hdf5)
+endif()
+if(EXISTS ${BASEDIR}/include/hdf)
+  set (INC_HDF ${BASEDIR}/include/hdf)
+endif()
+
+# Find GFTL 
+set(GFTL_IS_REQUIRED_ARG "REQUIRED" CACHE STRING "Argument in GFTL's find_package call")
+mark_as_advanced(GFTL_IS_REQUIRED_ARG)
+find_package(GFTL ${GFTL_IS_REQUIRED_ARG} CONFIG)
+
+# Find GFTL_SHARED
+set(GFTL_SHARED_IS_REQUIRED_ARG "" CACHE STRING "Argument in GFTL_SHARED's find_package call")
+mark_as_advanced(GFTL_SHARED_IS_REQUIRED_ARG)
+find_package(GFTL_SHARED ${GFTL_SHARED_IS_REQUIRED_ARG} CONFIG)
+
+# Find FARGPARSE
+set(FARGPARSE_IS_REQUIRED_ARG "" CACHE STRING "Argument in FARGPARSE's find_package call")
+mark_as_advanced(FARGPARSE_IS_REQUIRED_ARG)
+find_package(FARGPARSE ${FARGPARSE_IS_REQUIRED_ARG} CONFIG)
+
+# Find FLAP
+set(FLAP_IS_REQUIRED_ARG "REQUIRED" CACHE STRING "Argument in FLAP's find_package call")
+mark_as_advanced(FLAP_IS_REQUIRED_ARG)
+find_package(FLAP ${FLAP_IS_REQUIRED_ARG} CONFIG)
+# Set non-standard expected variables
+set (INC_FLAP ${FLAP_INCLUDE_DIRS})
```

Comment:
What are these being used for? There is nothing in the current MAPL 2.0 GCM that uses `INC_FLAP` or `LIB_FLAP`. FLAP is found via `FLAP::FLAP`.

Reply:
These are leftovers from before. I'll remove them.

Reply:
Fixed in 24097922a81f0a8e1d03ce658c6a8f0b4eedcfc1